coderd/externalauth/externalauth.go

```diff
@@ -396,10 +403,15 @@ func (c *DeviceAuth) ExchangeDeviceCode(ctx context.Context, deviceCode string)
 	if body.Error != "" {
 		return nil, xerrors.New(body.Error)
 	}
+	// If expiresIn is 0, then the token never expires.
+	expires := dbtime.Now().Add(time.Duration(body.ExpiresIn) * time.Second)
+	if body.ExpiresIn == 0 {
+		expires = time.Time{}
+	}
```

Comment:
This was another bug I found. If zero time was passed in, the token was considered immediately expired on our end.

Reply:
That's probably a big one. IIRC GitHub often just gives you tokens with no expiry.